### ISSUE TITLE
Feature: POMs should be automatically migrated to account for Fabric8 throwing errors for non-Docker modules

### ISSUE THREAD

**Issue by ewilkins-csi** [boozallen/aissemble#320]
## Description
Orphedomos was written to simply skip plugin execution if the packaging type of the POM was not `orphedomos`.  This allowed for configuring the plugin in the parent module of the Docker builds (i.e. in `<project>-docker/pom.xml`).  Fabric8 will throw an error in this instance.  To properly migrate downstream projects upgrading to 1.8, we need to account for this difference by ensuring any docker plugin configuration in non-docker modules is confined to a `pluginManagement` section, and ensure that all `docker-build` modules have the docker plugin directly added to their build.

## DOD
- [x] Migrated POMs with `pom` packaging have any Fabric8 docker plugin specification moved to an appropriate `pluginManagement` section
   - this includes within profiles, where the plugin configuration should be moved to a management section within the same profile
- [x] Migrated POMs with `docker-build` packaging have at least a bare bones Docker plugin config add to the main `build/plugins` section
- [x] These migrations run _after_ other Orphedomos-to-Fabric8 migrations
- [x] Archetype and MDA templates should be updated to contain these fixes by default
- [ ] Archetype/MDA changes should be merged to `dev`

### Test Upgrade to 1.8.1
 1. Create a new project from the 1.7 archetype
     ```sh
     mvn archetype:generate -B '-DarchetypeGroupId=com.boozallen.aissemble' \
                               '-DarchetypeArtifactId=foundation-archetype' \
                               '-DarchetypeVersion=1.7.0' \
                               '-DgroupId=org.test' \
                               '-Dpackage=org.test' \
                               '-DprojectGitUrl=test.org/test.git' \
                               '-DprojectName=Test Project' \
                               '-DartifactId=ots-320-upgrade' \
                               '-DdockerProjectRepositoryUrl=test-registry/' \
     && cd ots-320-upgrade
     ```
 2. Add [SparkPipeline.json](https://github.com/user-attachments/files/16965309/SparkPipeline.json) to _ots-320-upgrade-pipeline-models/src/main/resources/pipelines_
 3. Run the build and resolve manual actions until none are printed: `mvn clean install`
 4. Change the `build-parent` version in the root _pom.xml_ to **1.8.1-SNAPSHOT**
 5. Run the build and verify it is successful: `mvn clean install`
 6. Run the build with the CI profile and verify it is successful
     ```sh
     mvn clean install -Pci -pl :ots-320-upgrade-docker -amd
     ```
 7. Verify each docker module executes the `docker:build` goal during the build
 8. In _ots-320-upgrade-docker/pom.xml_ update the `docker-maven-plugin` within the first `pluginManagement` block by changing `skip` from `true` to `false`
 9. Run the build: `mvn clean install`
 10. Verify that the docker modules produced images
     ```sh
     docker images | grep ots-320
     ```

### Test 1.9
 1. Create a new project from the 1.9 snapshot archetype
     ```sh
     mvn archetype:generate -B '-DarchetypeGroupId=com.boozallen.aissemble' \
                               '-DarchetypeArtifactId=foundation-archetype' \
                               '-DarchetypeVersion=1.9.0-SNAPSHOT' \
                               '-DgroupId=org.test' \
                               '-Dpackage=org.test' \
                               '-DprojectGitUrl=test.org/test.git' \
                               '-DprojectName=Test Project' \
                               '-DartifactId=ots-320' \
                               '-DdockerProjectRepositoryUrl=test-registry/' \
     && cd ots-320
     ```
 2. Add [SparkPipeline.json](https://github.com/user-attachments/files/16965309/SparkPipeline.json) to _ots-320-pipeline-models/src/main/resources/pipelines_
 3. Run the build and resolve manual actions until none are printed: `mvn clean install`
 4. Run the docker builds with the CI profile and verify it is successful
     ```sh
     mvn clean install -Pci -pl :ots-320-docker -amd
     ```
 5. Verify each docker module executes the `docker:build` goal during the build
 6. In _ots-320-docker/pom.xml_ update the `docker-maven-plugin` within the first `pluginManagement` block by changing `skip` from `true` to `false`
 7. Run the build: `mvn clean install`
 8. Verify that the docker modules produced images (_note the image from the upgrade test will likely still be present – ensure the images without `-upgrade` are **also** present_)
     ```sh
     docker images | grep ots-320
     ```

**Comment by ewilkins-csi** [comment]
DoD with @csun-cpointe

**Comment by ewilkins-csi** [comment]
Migration and template/archetype changes done! (Thanks @csun-cpointe!)  Just need to ensure anything worth keeping from the migrations are picked over to branch off of dev and run through OTS.

**Comment by csun-cpointe** [comment]
OTS looks good ✅

**Comment by ewilkins-csi** [comment]
Changes merged into 1.8.1 branch with successful build.

**Comment by csun-cpointe** [comment]
Final test passed!!
